plugins/radar-android-google-places/src/main/java/org.radarbase.passive.google.places/GooglePlacesManager.kt

```diff
@@ -108,8 +107,14 @@ class GooglePlacesManager(service: GooglePlacesService, @get: Synchronized priva
         register()
         status = SourceStatusListener.Status.READY
         placesProcessor.start()
-        placeHandler.execute { if (!placesClientCreated) createPlacesClient() }
-        service.broadcaster.register(DEVICE_LOCATION_CHANGED) { _, _ ->
+        placeHandler.execute { if (!placesClientCreated) {
+            service.createPlacesClient()
+        }
+        if (!placesClientCreated && !Places.isInitialized()) {
+            updateApiKey(apiKey)
+        } }
+        updateConnected()
+        placesBroadcastReceiver = service.broadcaster?.register(DEVICE_LOCATION_CHANGED) { _, _ ->
```

Comment:
To properly unregister the receiver and allow GooglePlacesManager to be garbage collected.